src/data/classification/acute-inflammations-1.jl

```diff
@@ -6,8 +6,8 @@ register(DataDep(
   "http://archive.ics.uci.edu/ml/machine-learning-databases/acute/diagnosis.data",
   "5d0b61049992b8327aa847f916f3389ec08f74b43dbb3fc5222ac07185d7c88f",
   post_fetch_method=(path -> begin
-    # First 2 bytes are garbage
-    data = LegacyStrings.utf16(read(path)[3:end])
+    # File is UTF-16 and first 2 bytes are garbage
```

Comment:
Are the first 2 bytes garbage or are they a [BOM](https://en.wikipedia.org/wiki/Byte_order_mark)?

If it's the latter then you probably want to check if it's bigendian and if so call `ntoh.(...)` on the `UInt16` array before transcoding.

Reply:
i.e. do something like:
```jl
utf16 = reinterpret(UInt16, read(path))
if utf16[1] == 0xfffe # BOM indicates wrong endianness
    utf16 .= bswap.(utf16)
end
@views data = replace!(transcode(String, utf16[2:end]), ','=>'.')
```

Reply:
Thanks for taking care here, you are completely right. Luckily the file in question here is not complicated at all so it hasn't mattered:

```
35,5	no	yes	no	no	no	no	no
35,9	no	no	yes	yes	yes	yes	no
35,9	no	yes	no	no	no	no	no
36,0	no	no	yes	yes	yes	yes	no
36,0	no	yes	no	no	no	no	no
...
```

but I will incorporate in case anybody ever copies this code in future

Reply:
(Even if the file is purely ASCII, the BOM matters if the file was saved using the opposite endianness.   For example, the string `"no yes"` in UTF-16 would become `"渀漀 礀攀猀"` if you transcoded using the wrong endianness.)

Reply:
Resolved in 24e8bcfcbda6ec06c5f63507641762b32f66a7c5